### PR TITLE
SpeziLLM Remote OpenAI integration

### PR DESCRIPTION
**Pull request by philippzagar**
# SpeziLLM Remote OpenAI integration

## :recycle: Current situation & Problem
Currently, the module provides basic OpenAI integration, however, not in the SpeziLLM ecosystem.

## :gear: Release Notes 
- Add the `SpeziLLMOpenAI` target that provides an OpenAI integration on the basis of the SpeziLLM ecosystem.

## :books: Documentation
Added in-line docs + DocC articles + README

## :white_check_mark: Testing
Wrote basic UI test cases, manual testing

## :pencil: Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md):
- [X] I agree to follow the [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -26,11 +26,12 @@ let package = Package(
     dependencies: [
         .package(url: "https://github.com/MacPaw/OpenAI", .upToNextMinor(from: "0.2.4")),
         .package(url: "https://github.com/StanfordBDHG/llama.cpp", .upToNextMinor(from: "0.1.6")),
-        .package(url: "https://github.com/StanfordSpezi/Spezi", .upToNextMinor(from: "0.8.0")),
+        .package(url: "https://github.com/StanfordSpezi/Spezi", .upToNextMinor(from: "0.8.2")),
         .package(url: "https://github.com/StanfordSpezi/SpeziStorage", .upToNextMinor(from: "0.5.0")),
         .package(url: "https://github.com/StanfordSpezi/SpeziOnboarding", .upToNextMinor(from: "0.7.0")),
         .package(url: "https://github.com/StanfordSpezi/SpeziSpeech", .upToNextMinor(from: "0.1.1")),
-        .package(url: "https://github.com/StanfordSpezi/SpeziChat", .upToNextMinor(from: "0.1.1")),
+        // .package(url: "https://github.com/StanfordSpezi/SpeziChat", .upToNextMinor(from: "0.1.1")),
```
Comment (philippzagar) [inline Package.swift:33]: Note to ourselves: Switch back dependency as soon as PR got merged

File: README.md
Comment (PSchmiedmayer) [inline README.md:1]: Very nice examples! 🚀

File: Sources/SpeziLLM/Helpers/Chat+Append.swift
Comment (PSchmiedmayer) [inline Sources/SpeziLLM/Helpers/Chat+Append.swift:1]: Good extensions 👍

File: Sources/SpeziLLM/Tasks/LLMRunnerSetupTaskBuilder.swift [hunk 2]
```diff
@@ -7,52 +7,16 @@
...
-    
-    /// If declared, this will be called on the partial result of an 'if #available' block to allow the result builder to erase type information.
-    public static func buildLimitedAvailability(_ component: [any LLMRunnerSetupTask]) -> [any LLMRunnerSetupTask] {
-        component
-    }
-    
-    /// If declared, this will be called on the partial result from the outermost block statement to produce the final returned result.
-    public static func buildFinalResult(_ component: [any LLMRunnerSetupTask]) -> _LLMRunnerSetupTaskCollection {
-        _LLMRunnerSetupTaskCollection(runnerSetupTasks: component)
+public enum LLMRunnerSetupTaskBuilder: DependencyCollectionBuilder {
```
Comment (PSchmiedmayer) [inline Sources/SpeziLLM/Tasks/LLMRunnerSetupTaskBuilder.swift:17]: Nice to see the new Spezi additions being used here 👍

**Review by PSchmiedmayer — APPROVED** [review]
Very nice work @philippzagar! As we discussed all the different elements in our last meetings I don't have change requests for this PR. 

Enjoy your holidays and thanks for all the work here! 🚀

**Comment by philippzagar** [comment]
@PSchmiedmayer As soon as https://github.com/StanfordSpezi/SpeziChat/pull/5 is merged and tagged, feel free to trigger another pipeline run (as the specified `SpeziChat` dependency isn't tagged yet to `0.1.2`) and then merge the PR! 🚀

**Comment by PSchmiedmayer** [comment]
@philippzagar Sounds great; I just tagged the 0.1.2 release in Spezi Chat 🚀

**Comment by philippzagar** [comment]
@PSchmiedmayer Just triggered another pipeline run, thanks for tagging SpeziChat!

**Pull request merged**